skimage/segmentation/slic_superpixels.py

```diff
@@ -1,17 +1,111 @@
+import warnings
+import functools
 import collections as coll
 import numpy as np
 from scipy import ndimage as ndi
+from scipy.spatial.distance import pdist, squareform
 
+from ._slic import (_slic_cython, _enforce_label_connectivity_cython)
 from ..util import img_as_float, regular_grid
-from ..segmentation._slic import (_slic_cython,
-                                  _enforce_label_connectivity_cython)
 from ..color import rgb2lab
 
 
+def _get_mask_centroids(mask, n_centroids, spacing=None):
+    """Find regularly spaced centroids on a mask.
+
+    Parameters
+    ----------
+    mask : 3D ndarray
+        The mask where the centroids must be positioned.
+    n_centroids : int
+        The number of centroids to be returned.
+    spacing : sequence of same, optional,
+        Spacing of elements along each dimension.
+
+    Returns
+    -------
+    centroids : 2D ndarray
+        The coordinates of the centroids with shape (n_centroids, 3).
+    steps : 1D ndarray
+        The approximate distance between two seeds in all dimensions.
+
+    """
```

Comment:
Would you mind explaining the sampling strategy in more detail here? I don't actually understand what this function is doing. (I do intend to read the paper, but I think for our users, it would be nice if the strategy was explained somewhere. Probably in the argument for `mask=` in the public-facing `slic` function.

Reply:
I added some comments to the `_get_mask_centroids` function explaining the algorithm and modified the doc of the `mask` argument.
I also added a note in the doc explaining the behavior of the function according to the `spacing` argument value introduced by the [suggestion](https://github.com/scikit-image/scikit-image/pull/3850#discussion_r289618688) of @emmanuelle